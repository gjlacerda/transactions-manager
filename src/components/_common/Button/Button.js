import React from 'react'
import styled from 'styled-components'
import {
  colorApp,
  colorAppLight,
  colorAppDark,
  colorWhite,
  colorGray,
  colorGrayDark,
  colorBlack,
} from 'utils/color'
import { borderRadius } from 'utils/border'
import { inputShadow } from 'utils/shadow'
import { transitionInput } from 'utils/transition'
import { opacityDisabled } from 'utils/opacity'

const Container = styled.button`
  padding: 10px 20px;
  background: ${props => (props.primary ? colorApp : colorGray)};
  color: ${props => (props.primary ? colorWhite : colorBlack)};
  outline: none;
  border: none;
  border-radius: ${borderRadius};
  cursor: pointer;
  box-shadow: ${inputShadow};
  transition: ${transitionInput('background-color')};
  opacity: ${props => props.disabled ? opacityDisabled : 1};

  &:hover {
    background-color: ${props => (props.primary ? colorAppDark : colorGrayDark)};
  }

  &:active {
    background-color: ${props => (props.primary ? colorAppLight : colorGray)};
  }
`

export default ({
  primary,
  onClick,
  type,
  disabled,
  children,
}) => (
  <Container
    type={type}
    primary={primary}
    disabled={disabled}
    onClick={onClick}
  >
    {children}
  </Container>
)
