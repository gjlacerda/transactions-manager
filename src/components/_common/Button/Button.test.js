import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Button from './Button'

describe('Button component', () => {
  it('deve ser igual ao snapshot', () => {
    const tree = renderer.create(
      <Button />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('deve ser igual ao snapshot quando passado o parâmetro (fullWidth)', () => {
    const tree = renderer.create(
      <Button fullWidth />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('deve chamar a função onClick passada por parâmetro', () => {
    const spy = jest.fn()
    const wrapper = shallow(<Button onClick={spy} />)
    wrapper.simulate('click')
    expect(spy).toHaveBeenCalled()
  })
})
