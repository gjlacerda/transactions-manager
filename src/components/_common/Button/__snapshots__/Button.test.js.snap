// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component deve ser igual ao snapshot 1`] = `
.c0 {
  padding: 10px 20px;
  background: #e0e0e0;
  color: #555;
  outline: none;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 3px 1px -2px rgba(0,0,0,0.12);
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  opacity: 1;
}

.c0:hover {
  background-color: #b3b3b3;
}

.c0:active {
  background-color: #e0e0e0;
}

<button
  className="c0"
/>
`;

exports[`Button component deve ser igual ao snapshot quando passado o par√¢metro (fullWidth) 1`] = `
.c0 {
  padding: 10px 20px;
  background: #e0e0e0;
  color: #555;
  outline: none;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 3px 1px -2px rgba(0,0,0,0.12);
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  opacity: 1;
}

.c0:hover {
  background-color: #b3b3b3;
}

.c0:active {
  background-color: #e0e0e0;
}

<button
  className="c0"
/>
`;
